### PR TITLE
Fix modules

### PR DESCRIPTION
**Pull request by koops76**
Ping @RReverser @cowlicks @Hainish.

### CODE REVIEW

**Comment by RReverser** [comment]
Thanks for fixing! Not sure all the duplicated wrappers are worth it TBH, especially given that this still doesn't fix `rules.js` being not directly require-able from Node, still works only in extension unless you do some global patching.

**Comment by koops76** [comment]
Just copy `rules.js` and put a `background.js` shim in the same folder.

**Comment by RReverser** [comment]
No, I already explained in previous PR why it doesn't work that way... `background.js` uses Chrome-specific APIs even at top level, and it's not intended to be executed from Node.js. `rules.js` can and should be just a self-contained module instead of relying on any browser-specific bits.

**Comment by koops76** [comment]
```js
'use strict'

exports = {
  enableMixedRulesets: false,
  urlBlacklist: {
    has () { return false }
  }
}
```

**Comment by koops76** [comment]
That's the shim.

**Comment by koops76** [comment]
@RReverser

**Comment by RReverser** [comment]
Which means the end script has to provide own shims just to require rewriter? Even for things like https://github.com/EFForg/https-everywhere/pull/12558 I would have to keep separate shim and preload it into global before requiring rules.js? I don't really see why this complication is needed when a much simpler solution can be done that wouldn't require any extra effort on consumption side...

**Comment by koops76** [comment]
Hmm, how?

**Comment by koops76** [comment]
I guess a copy of `rules.js` and the shim can be copied into a npm package and kept somewhere in the repo. Scripts may just use it.

**Comment by RReverser** [comment]
Of course, anything can be shimmed, worked around, patched, hacked etc. on consumer side, but why not make it simple in the first place when it's as easy as moving two variables between files? All I want is to be able to launch Node.js, type `require('./rules')` and have it already working.

> Hmm, how?

As wrote here: https://github.com/EFForg/https-everywhere/pull/12586#issuecomment-330318222

Basically make rules.js (or, ideally, each script) independent, just providing APIs for options that can be modified externally.

**Comment by koops76** [comment]
Okay. This PR will fix two bugs then.

**Comment by cowlicks** [comment]
Hey @koops76 @RReverser  I'm working on fixing this, so please hold off for a bit until I can get it out.

**Comment by koops76** [comment]
@RReverser Added standalone `rules.js` support.

File: chromium/rules.js [hunk 1]
```diff
@@ -248,7 +252,7 @@
       var platform = ruletag["platform"]
       if (platform) {
         default_state = false;
-        if (platform == "mixedcontent" && background.enableMixedRulesets) {
+        if (platform == "mixedcontent" && background.enableMixedRulesets.enable) {
```
Comment (RReverser) [inline chromium/rules.js]: Shouldn't this be without `background.`?

File: chromium/background.js [hunk 2]
```diff
@@ -61,9 +61,9 @@
   all_rules = new rules.RuleSets(ls);
 
   // Allow users to enable `platform="mixedcontent"` rulesets
-  var enableMixedRulesets = false;
+  rules.enableMixedRulesets = false;
```
Comment (RReverser) [inline chromium/background.js]: Missed `.enable`?

File: chromium/rules.js [hunk 3]
```diff
@@ -252,7 +250,7 @@
       var platform = ruletag["platform"]
       if (platform) {
         default_state = false;
-        if (platform == "mixedcontent" && background.enableMixedRulesets.enable) {
+        if (platform == "mixedcontent" && enableMixedRulesets.enable) {
```
Comment (RReverser) [inline chromium/rules.js]: Btw why do you need this indirection and not just `enableMixedRulesets` being a boolean directly?

Comment (koops76) [reply chromium/rules.js]: This won't work since it needs to be the same object.

Comment (koops76) [reply chromium/rules.js]: This can be done with `get()` and `set()` methods instead.

Comment (RReverser) [reply chromium/rules.js]: Or you can just use `exports.enableMixedRulesets`. Then it will be same inside and outside.

**Comment by cowlicks** [comment]
@koops76  would you mind marking this as **WIP**  in the title so we know not to review it?

**Comment by koops76** [comment]
@cowlicks Wait until Travis will finish.

**Comment by cowlicks** [comment]
@koops76 oops sorry. I meant to comment on your ES6 imports PR. Can you mark that is WIP so we know not to review it.

Also, I'm much more in favor of using #12556 to fix this.

**Comment by koops76** [comment]
@cowlicks Done.

**Comment by koops76** [comment]
@cowlicks It's closed, and if you fix all conflicts and merge it anyway, it will cause conflicts for this PR.

**Comment by cowlicks** [comment]
Module fixes are being done in #12627 is this still needed?

**Pull request closed without merging**